tests/CommunityToolkit.Aspire.Hosting.SurrealDb.Tests/SurrealDbFunctionalTests.cs

```diff
@@ -244,6 +247,94 @@ public async Task VerifyWaitForOnSurrealDbBlocksDependentResources()
 
         await app.StopAsync();
     }
+    
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyWithInitFiles()
+    {
+        // Creates a script that should be executed when the container is initialized.
+    
+        using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, BackoffType = DelayBackoffType.Linear, Delay = TimeSpan.FromSeconds(2), ShouldHandle = new PredicateBuilder().Handle<SurrealDbException>() })
+            .Build();
+    
+        var initDirPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+    
+        Directory.CreateDirectory(initDirPath);
+        
+        var initFilePath = Path.Combine(initDirPath, "init.surql");
+    
+        var surrealNsName = "ns1";
+        var surrealDbName = "db1";
+        
+        try
+        {
+            await File.WriteAllTextAsync(
+                initFilePath, 
+        $"""
+                USE NS {surrealNsName};
+                USE DB {surrealDbName};
+                
+                CREATE car SET Brand = "BatMobile";
+                """, 
+                cts.Token
+            );
+    
+            using var builder = TestDistributedApplicationBuilder.Create(testOutputHelper);
+    
+            var surrealServer = builder
+                .AddSurrealServer("surreal")
+                .WithInitFiles(initFilePath);
+    
+            var ns = surrealServer.AddNamespace(surrealNsName);
+            var db = ns.AddDatabase(surrealDbName);
+    
+            using var app = builder.Build();
+    
+            await app.StartAsync(cts.Token);
+                
+            var rns = app.Services.GetRequiredService<ResourceNotificationService>();
+            
+            await rns.WaitForResourceAsync(db.Resource.Name, KnownResourceStates.Running, cts.Token);
+    
+            var hb = Host.CreateApplicationBuilder();
+    
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(cts.Token)
+            });
+    
+            hb.AddSurrealClient(db.Resource.Name);
+    
+            using var host = hb.Build();
+    
+            await host.StartAsync(cts.Token);
+
+            // Wait until the database is available
+            await rns.WaitForResourceHealthyAsync(db.Resource.Name, cts.Token);
+    
+            await pipeline.ExecuteAsync(async token =>
+            {
+                var client = host.Services.GetRequiredService<SurrealDbClient>();
+
+                var cars = (await client.Select<Car>(Car.Table, token)).ToList();
+                Assert.True(cars.Count == 1);
+                Assert.Equal("BatMobile", cars[0].Brand);
```

Comment:
```suggestion
                var car = Assert.Single(cars);
                Assert.Equal("BatMobile", car.Brand);
```

You can use the xunit asserts to return values and access them in the next assert